boards/shields/waveshare_epaper/waveshare_epaper_gdew042t2-p.overlay

```diff
@@ -0,0 +1,95 @@
+/*
```

Comment:
You need additional changes to use it properly as a shield. There is also a PR that cleans it up https://github.com/zephyrproject-rtos/zephyr/pull/48791.

```diff
diff --git a/boards/shields/waveshare_epaper/Kconfig.defconfig b/boards/shields/waveshare_epaper/Kconfig.defconfig
index 10263e88bf2..b60e593e8cd 100644
--- a/boards/shields/waveshare_epaper/Kconfig.defconfig
+++ b/boards/shields/waveshare_epaper/Kconfig.defconfig
@@ -5,7 +5,8 @@
 # SPDX-License-Identifier: Apache-2.0
 #
 
-if SHIELD_WAVESHARE_EPAPER_GDEH029A1 || SHIELD_WAVESHARE_EPAPER_GDEH0213B1 || SHIELD_WAVESHARE_EPAPER_GDEH0213B72 || SHIELD_WAVESHARE_EPAPER_GDEW075T7 || SHIELD_WAVESHARE_EPAPER_GDEH0154A07 || SHIELD_WAVESHARE_EPAPER_GDEW042T2
+if SHIELD_WAVESHARE_EPAPER_GDEH029A1 || SHIELD_WAVESHARE_EPAPER_GDEH0213B1 || SHIELD_WAVESHARE_EPAPER_GDEH0213B72 || SHIELD_WAVESHARE_EPAPER_GDEW075T7 || SHIELD_WAVESHARE_EPAPER_GDEH0154A07 || SHIELD_WAVESHARE_EPAPER_GDEW042T2 || SHIELD_WAVESHARE_EPAPER_GDEW042T2_P
+
 
 
 if DISPLAY
diff --git a/boards/shields/waveshare_epaper/Kconfig.shield b/boards/shields/waveshare_epaper/Kconfig.shield
index 11b5567d7ed..863059e5784 100644
--- a/boards/shields/waveshare_epaper/Kconfig.shield
+++ b/boards/shields/waveshare_epaper/Kconfig.shield
@@ -18,3 +18,6 @@ config SHIELD_WAVESHARE_EPAPER_GDEH0154A07
 
 config SHIELD_WAVESHARE_EPAPER_GDEW042T2
        def_bool $(shields_list_contains,waveshare_epaper_gdew042t2)
+
+config SHIELD_WAVESHARE_EPAPER_GDEW042T2_P
+       def_bool $(shields_list_contains,waveshare_epaper_gdew042t2-p)
```

Reply:
Done.